langserver/completion.go

```diff
@@ -102,9 +103,33 @@ func (s *server) completeMetricName(ctx context.Context, completions *[]protocol
 	var allNames model.LabelValues
 
 	if api != nil {
-		var err error
+		// isCompatible checks if language server has access to new metadata APIs.
+		var isCompatible = false // nolint:ineffassign
+
+		requiredVersion := semver.MustParse("2.15.0")
+		presentVersion, err := semver.New(s.PrometheusVersion)
+
+		switch {
+		case err != nil && !s.headless:
+			break
+		case s.headless:
+			isCompatible = true
+		default:
+			isCompatible = presentVersion.GTE(requiredVersion)
+		}
+
+		if !isCompatible {
+			allNames, _, err = api.LabelValues(ctx, "__name__")
+		} else {
+			metadata, err := api.Metadata(ctx, "", "")
```

Comment:
In this case we can keep  `metadata` as is. Better convert `allNames` in the case above to a slice of the same type.

Reply:
We can then use `metadata` to populate the `Documentation` field of the returned `completionItem`s.

Reply:
What will be format of documentation string?

Reply:
Are you talking of [this line](https://github.com/prometheus-community/promql-langserver/blob/723971df19fc2b64d312f2fc2bfe79b844fb9536/internal/vendored/go-tools/lsp/protocol/tsprotocol.go#L503)?

Reply:
> What will be format of documentation string?

Maybe the `help` text for the `Documentation` field and the `type` for the `Detail` field.

I haven't seen the `unit` metadata field being used in the wild so much, so I wouldn't mind ignoring it, but if you have a idea about how to include it, go ahead.

> Are you talking of this line?

Yes.